### PR TITLE
Upgraded to android gradle plugin 3.0

### PR DESCRIPTION
**Pull request by deepankarb**
* Upgraded android gradle plugin to 3.0.0-beta4.
* Upgraded some dependencies to their latest version.
* Bumped compileSdk and targetSdk to android O (26).

### CODE REVIEW

**Comment by ribotaccounts** [comment]
Can one of the admins verify this patch?

File: app/build.gradle [hunk 1]
```diff
@@ -5,17 +5,23 @@ apply plugin: 'com.github.triplet.play'
...
         testInstrumentationRunner "${applicationId}.runner.RxAndroidJUnitRunner"
         versionCode 1000
         // Major -> Millions, Minor -> Thousands, Bugfix -> Hundreds. E.g 1.3.72 == 1,003,072
         versionName '0.1.0'
+
+        javaCompileOptions {
+            annotationProcessorOptions {
+                includeCompileClasspath false
+            }
+        }
```
Comment (RichSuther) [inline app/build.gradle]: I dont think this is needed post Butterknife 8.6?

See: https://github.com/JakeWharton/butterknife/issues/908

**Comment by ribotaccounts** [comment]
Can one of the admins verify this patch?

**Comment by deepankarb** [comment]
I did not intend to close this. I wonder how that happened. Anyways this needs some more work since AS 3.0 is stable now. Will send in another PR.

**Pull request closed without merging**